deeplearning4j-ui-parent/deeplearning4j-ui-model/src/main/java/org/deeplearning4j/ui/stats/BaseStatsListener.java

```diff
@@ -222,59 +228,71 @@ public void onEpochEnd(Model model) {
     @Override
     public void onForwardPass(Model model, List<INDArray> activations) {
         int iterCount = getModelInfo(model).iterCount;
-        if (storeActivations() && (iterCount == 0 || iterCount % updateConfig.reportingFrequency() == 0)) {
+        if (calcFromActivations() && (iterCount == 0 || iterCount % updateConfig.reportingFrequency() == 0)) {
```

Comment:
Is this handled on a separate thread similar/same to as before?

Reply:
No, it's handled in the current thread, same as before. It just does it when the arrays are available, rather than duping + calculating all at one later.